sfge/src/main/java/com/salesforce/graph/ops/ApexPathUtil.java

```diff
@@ -232,5 +243,45 @@ private enum Direction {
         BACKWARD;
     }
 
+    /**
+     * Object generated during path retrieval. Includes information about all accepted paths, as well as all reasons that paths were rejected.
+     */
+    public static final class ApexPathRetrievalSummary {
```

Comment:
This class allows `getPaths()` to easily return the same type of object regardless of whether it attempted path expansion.